### PR TITLE
[Strings3] Update documentation and codebase with new string representations of X, Y, Z and Identity

### PR DESCRIPTION
**Pull request by Qottmann**
Updating docs and codebase with `qml.X/Y/Z/I(*)` instead of `qml.PauliX/Y/Z(*)` and `qml.Identity(*)` following https://github.com/PennyLaneAI/pennylane/pull/5116 and updating composite operation outputs in docs according to https://github.com/PennyLaneAI/pennylane/pull/5138/

ToDo

- [x] Improve https://docs.pennylane.ai/en/stable/introduction/operations.html#qubit-operators
- [x] Update outputs in docs with PauliX(wires=*)
- [x] Update outputs in docs with SProds and other composites (i.e. with extra spaces and removing unnecessary brackets)
- [x] manually check & update all intro pages
- [x] manually check & update pulse docs
- [x] All tests pass
- [x] changelog

### CODE REVIEW

File: pennylane/ops/op_math/exp.py [hunk 1]
```diff
@@ -78,13 +78,13 @@ def exp(op, coeff=1, num_steps=None, id=None):
     This symbolic operator can be used to make general rotation operators:
 
     >>> x = np.array(1.23)
-    >>> op = qml.exp( qml.PauliX(0), -0.5j * x)
+    >>> op = qml.exp( qml.X(0), -0.5j * x)
```
Comment (trbromley) [inline pennylane/ops/op_math/exp.py:81]:
```suggestion
    >>> op = qml.exp(qml.X(0), -0.5j * x)
```

**Review by trbromley — APPROVED** [review]
Thanks @Qottmann! This was from a find-and-replace right? :laughing: 

My main reflection is that, given we are pointing everyone to `X`, `Y`, `Z`, we should probably improve the docs for them ([e.g.](https://docs.pennylane.ai/en/latest/code/api/pennylane.X.html)) - maybe making `PauliX` the alias instead.

**Comment by Qottmann** [comment]
Excellent point! I went the route of renaming classes PauliXYZ -> X and then making PauliX, PauliY, and PauliZ the aliases. This way we still have `isinstance(qml.X, PauliX)` be true.

However, a lot of the codebase (and presumably plugins) rely on the `__class__.__name__` as well as the `name` property. I first tried to "just" update the codebase accordingly but this turned out really messy really soon (perhaps due to my ignorance on some parts of the codebase), so instead I now went with the solution to just overwrite those properties in the classes, so all the mechanics behind the scenes stay in tact.

It seems to work, the failing test is stochastic I believe (passes locally)

**Comment by Qottmann** [comment]
Excellent point!

A lot of the codebase seems to rely on checking not just `isinstance(op, PauliX)` but also directly checking the name of the class or instance. I went with the solution to have `X/Y/Z/I` be the main classes and overwrite the class name. `PauliX/Y/Z/Identity` are now just aliases. This way the mechanics in the codebase and any plugins remain functioning.

I also tried subclassing but it turned out difficult to have both `isinstance(qml.PauliX(0), qml.X)` and `isinstance(qml.X(0), qml.PauliX)`. I also tried with a metaclass to overwrite the `__instancecheck__` but I couldnt get it to work (happy to discuss though)

**Review by albi3ro — APPROVED** [review]
Well that was painful to get through...

**Comment by Qottmann** [comment]
Now:

image

and 

image

Sounds good?

**Pull request merged**